lib/ask_web/controllers/respondent_controller.ex

```diff
@@ -563,6 +564,38 @@ defmodule AskWeb.RespondentController do
     |> min(100.0)
   end
 
+  defp percentages(%{started_at: nil}), do: %{}
+
+  defp percentages(survey) do
+    %{
+      success_rate: cleanup_repetitive_percentages(Survey.success_rate_history(survey))
+    }
+  end
+
+  # Cleanup repetitive values, by skipping any repetitive value except for the
+  # first and last, because they don't contribute anything to the graph
+  # (straight line).
+  #
+  # For example take A, B, C, D consecutive dates, each at value 5.0 then we
+  # only keep A and D and skip B and C entirely.
+  #
+  # TODO: consider moving to RespondentView.
```

Comment:
Maybe this ought to be in RespondentView? The controller is doing the job in the controller, so I made it here.

Reply:
Please let me take advantage of your dilemma here. How would you decide this? Which considerations would you take into account to decide? Thanks!!!

Reply:
I didn't move the cleanup because the controller already cleans up the similar `cumulative_percentages` to the view, because it does that while doing the sum (not as a separate step), so it felt proper to send it the cleaned up array, even though logically it should be in the view :shrug: